psi4/src/psi4/libscf_solver/rohf.cc

```diff
@@ -426,8 +425,22 @@ void ROHF::form_F() {
     }
 }
 
-void ROHF::form_C() {
-    soFeff_->diagonalize(Ct_, epsilon_a_);
+void ROHF::form_C(double shift) {
```

Comment:
Orbital energies in ROHF aren't even well-defined, so it's not obvious to me how level-shifted ROHF would work. I think the canonical reference on this is [here](https://www.tandfonline.com/doi/abs/10.1080/00268977400102171), but Emory doesn't seem to have online access to Mol. Phys., so I can't access it right now.

Reply:
Now that I've been able to get that paper: the ROHF level shifting approach they recommend (eq. 15) is to apply level shifter `alpha` to the open orbitals and level-shifter `alpha+beta` to the virtual orbitals. They don't do an exhaustive study of optimal level shifts, but they use `beta`=`alpha`.

Reply:
But what are these orbitals in the code itself?

Reply:
I don't understand the question. Are you asking how to obtain the orbital spaces? The [`form_F` function documentation](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libscf_solver/rohf.cc#L361) establishes how the ROHF Fock matrix is arranged, and that `doccpi` and `soccpi` work as you would naively expect.

If that doesn't answer the question, please rephrase.

Reply:
Oh, okay. I've changed the code accordingly. I use the same level shift (`shift`) for virtual orbitals as in RHF, and thereby shift the open-shell orbitals by `shift/2`. However, I'm not sure it is working properly; e.g. the example calculation of Guest and Saunders
```
molecule {
units bohr
0 2
B
O 1 2.277
}

set basis pcseg-1
set reference rohf
#set reference uhf
set diis false
set level_shift 0.2
set level_shift_convergence 1e-3
energy('scf')
```

does not converge.

Reply:
So what I don't understand is the difference between `Ca_` and `Ct_`.

Reply:
OK, solved; ROHF works.